build.gradle.kts

```diff
@@ -31,81 +31,47 @@ detekt {
     ignoredFlavors = listOf("production")
 }
 
-jacoco {
-    toolVersion = libs.versions.jacoco.get()
-}
-
-val fileGenerated = listOf(
-    "android/**/*.*",
-    "**/R.class",
-    "**/R\$*.class",
-    "**/*\$ViewBinder*.*",
-    "**/*\$InjectAdapter*.*",
-    "**/*Injector*.*",
-    "**/BuildConfig.*",
-    "**/Manifest*.*",
-    "**/*_ViewBinding*.*",
-    "**/*_Factory*.*",
-    "**/app/ui/screens/**/*DiffCallback*.*",
-    "**/*Test*.*",
-    // navigation component
-    "**/*FragmentArgs*",
-    "**/*FragmentDirections*",
-    "**/FragmentNavArgsLazy.kt",
-    "**/*Fragment*navArgs*",
-    "**/screens/common/StartFragment.*",
-    // kotlin enum Creator
-    "**/*\$Creator*"
-)
-
-val packagesExcluded = listOf(
-    "co/nimblehq/extensions/app/**",
-    "com/bumptech/glide"
-)
-
-val fileFilter = fileGenerated + packagesExcluded
-
-tasks.register<JacocoReport>("jacocoTestReport") {
-    group = "Reporting"
-    description = "Generate Jacoco coverage reports for Debug build"
-
-    dependsOn(":app:testDebugUnitTest")
-    dependsOn(":common-ktx:testDebugUnitTest")
-
-    classDirectories.setFrom(
-        fileTree("${project.rootDir}/app/build/intermediates/javac/debug/classes") {
-            exclude(fileFilter)
-        },
-        fileTree("${project.rootDir}/common-ktx/build/intermediates/javac/debug/classes") {
-            exclude(fileFilter)
-        },
-        fileTree("${project.rootDir}/app/build/tmp/kotlin-classes/debug") {
-            exclude(fileFilter)
-        },
-        fileTree("${project.rootDir}/common-ktx/build/tmp/kotlin-classes/debug") {
-            exclude(fileFilter)
+subprojects {
+    apply(plugin = "org.jetbrains.kotlinx.kover")
+    configure<KoverProjectExtension> {
+        reports {
+            filters {
+                excludes {
+                    androidGeneratedClasses()
+                    annotatedBy(
+                        "androidx.annotation.VisibleForTesting"
+                    )
+                    classes(
+                        // View Binding & Data Binding
+                        "**/*_ViewBinding*",
+                        "**/*_Factory*",
+                        "**/*\$ViewBinder*",
+                        "**/*\$InjectAdapter*",
+                        "**/*Injector*",
+                        "*.*_ComponentTreeDeps*",
+                        "*.*_HiltComponents*",
+                        "*.*_MembersInjector*",
+                        "*\$InstanceHolder",
+                        // Navigation Component
+                        "**/*FragmentArgs*",
+                        "**/*FragmentDirections*",
+                        "**/FragmentNavArgsLazy*",
+                        "**/*Fragment*navArgs*",
+                        "**/screens/common/StartFragment*",
```

Comment:
Side note: We will need to remove/update the exclusive files later on as there are many files out of dated.